### PR TITLE
Make ``a = [a, b]`` array idiom work

### PR DESCRIPTION
**Pull request by czgdp1807**
Closes https://github.com/lfortran/lfortran/issues/2026

### CODE REVIEW

File: tests/reference/asr-array3-7b1ab3b.stdout [hunk 1]
```diff
@@ -123,7 +123,7 @@
                                 )
                             )]
                             (Array
-                                (Integer 4)
+                                (Real 4)
```
Comment (certik) [inline tests/reference/asr-array3-7b1ab3b.stdout:126]: Is this change correct?

**Review by certik — APPROVED** [review]
I think that this is fine. Thanks for implementing it. I left just one question, if the change is correct, then you can merge this.

File: tests/reference/asr-array3-7b1ab3b.stdout (hunk 1 above)
Comment (czgdp1807) [reply tests/reference/asr-array3-7b1ab3b.stdout:126]: Yes. Casting is needed from Integer to Real. Earlier it was incorrect.

**Comment by certik** [comment]
Awesome, thanks!

**Pull request merged**